deep_qa/tensors/backend.py

```diff
@@ -17,31 +18,20 @@ def switch(cond, then_tensor, else_tensor):
     tensorflow's where, we can exactly retrieve this functionality.
     """
 
-    if K.backend() == 'tensorflow':
-        import tensorflow as tf
-        cond_shape = cond.get_shape()
-        input_shape = then_tensor.get_shape()
-        if cond_shape[-1] != input_shape[-1] and cond_shape[-1] == 1:
-            # This happens when the last dim in the input is an embedding dimension. Keras usually does not
-            # mask the values along that dimension. Theano broadcasts the value passed along this dimension,
-            # but TF does not. Using K.dot() since cond can be a tensor.
-            cond = K.dot(tf.cast(cond, tf.float32), tf.ones((1, input_shape[-1])))
-        return tf.where(tf.cast(cond, dtype=tf.bool), then_tensor, else_tensor)
-    else:
-        import theano.tensor as T
-        return T.switch(cond, then_tensor, else_tensor)
-
-
+    cond_shape = cond.get_shape()
+    input_shape = then_tensor.get_shape()
+    if cond_shape[-1] != input_shape[-1] and cond_shape[-1] == 1:
+        # This happens when the last dim in the input is an embedding dimension. Keras usually does not
+        # mask the values along that dimension. Theano broadcasts the value passed along this dimension,
+        # but TF does not. Using K.dot() since cond can be a tensor.
+        cond = K.dot(tf.cast(cond, tf.float32), tf.ones((1, input_shape[-1])))
+    return tf.where(tf.cast(cond, dtype=tf.bool), then_tensor, else_tensor)
+    
 def cumulative_sum(tensor, axis=-1):
```

Comment:
This says "again" because I had an earlier comment that got erased when I refreshed the page at one point.  This issue appears also where there's a function called `my_keras_cumsum` in one of the files you touched (maybe one of the encoders?).